### PR TITLE
refactor(udf): many improves to udf system

### LINKED ISSUE #807: feat: generate udf doc for `RegisterAlias` as well
We might need to refactor .doc() interface.
Now the function documenting is a little bit confusing, especially when the developer alias a registered function.
We should find a way to document the alias function.

_Originally posted by @jingchen2222 in https://github.com/4paradigm/OpenMLDB/pull/791#discussion_r759592122_

### LINKED ISSUE #2702: integers unsupported for RADIANS
image

we should support integers as well

### LINKED ISSUE #2895: implicitly convert binary operands type for `nvl`
E.g. 

`nvl(count(col) over w , 123)` will report error:

```
W1221 11:43:44.691177 1204308160 resolve_fn_and_attrs.cc:235] Resolve function 'nvl' failed, fallback to legacy: Fail to resolve expr udf: Default value should take same type with input, expect int64 but get int32                         
    (At /Users/danielace/Git/4paradigm/OpenMLDB/hybridse/src/udf/udf_registry.h:91)                                                                                                                                                           
    (At /Users/danielace/Git/4paradigm/OpenMLDB/hybridse/src/udf/udf_registry.cc:59)                                                                                                                                                          
    (Caused by) Fail to resolve expr udf: Default value should take same type with input, expect int64 but get int32
```
`count` returns int64, not compatible with 123 (int32 type)

### PR DESCRIPTION
**Pull request by aceforeverd**
- more supported type for `radians`, fixes #2702 
- compatible data type for Cond like exprs, fixes #2895
- improves to udf doc gen:
  - aliased function doc, fixes #807

### CODE REVIEW

File: hybridse/src/udf/udf_registry.h [hunk 1]
```diff
@@ -282,6 +282,12 @@ struct UdfLibraryEntry {
 
     // record whether always return list
     bool always_return_list = false;
+
+    // canonical funtion name
+    std::string fn_name_;
+
+    // alias name to the function
+    std::set<std::string> alias_;
```
Comment (aceforeverd) [inline hybridse/src/udf/udf_registry.h]: cpplint

File: hybridse/src/udf/udf_registry.h [hunk 2]
```diff
@@ -282,6 +283,12 @@ struct UdfLibraryEntry {
 
     // record whether always return list
     bool always_return_list = false;
+
+    // canonical funtion name
+    std::string fn_name_;
```
Comment (dl239) [inline hybridse/src/udf/udf_registry.h]: rm the underline at the end of variable

File: hybridse/src/udf/default_udf_library.cc [hunk 3]
```diff
@@ -31,6 +31,7 @@
 #include "udf/containers.h"
 #include "udf/udf.h"
 #include "udf/udf_registry.h"
+#include "udf/default_defs/expr_def.h"
```
Comment (dl239) [inline hybridse/src/udf/default_udf_library.cc:34]: reorder the header

File: hybridse/src/udf/default_udf_library.cc [hunk 4]
```diff
@@ -1631,7 +1631,9 @@ void DefaultUdfLibrary::InitMathUdf() {
                 SELECT RADIANS(90.0);
                 --output 1.570796326794896619231
             @endcode
-            @since 0.6.0)");
+
+            @since 0.6.0)")
```
Comment (dl239) [inline hybridse/src/udf/default_udf_library.cc:1635]: new added types are not supported in `0.6.0`. update the since ?

**Review by dl239 — APPROVED** [review]

File: hybridse/src/udf/default_udf_library.cc (hunk 4 above)
Comment (aceforeverd) [reply hybridse/src/udf/default_udf_library.cc:1635]: no, I fix this as bug, so version number not change. If changed all signatures are updated to the new version.

**Review by vagetablechicken — APPROVED** [review]

**Pull request merged**